core/src/data_sources/data_source.rs

```diff
@@ -957,64 +957,85 @@ impl DataSource {
                 chunks
             }
             Some(q) => {
-                let r = EmbedderRequest::new(
-                    self.config.provider_id,
-                    &self.config.model_id,
-                    vec![&q],
-                    self.config.extras.clone(),
-                );
-                let v = r.execute(credentials).await?;
-                assert!(v.len() == 1);
-
-                utils::done(&format!(
-                    "DSSTAT Finished embedding query: duration={}ms",
-                    utils::now() - time_embedding_start,
-                ));
-
-                // Construct the filters for the search query if specified.
-                let f = build_qdrant_filter(&filter);
-
-                let time_search_start = utils::now();
-                let results = qdrant_client
-                    .search_points(&qdrant::SearchPoints {
-                        collection_name: self.qdrant_collection(),
-                        vector: v[0].vector.iter().map(|v| *v as f32).collect::<Vec<f32>>(),
-                        filter: f,
-                        limit: top_k as u64,
-                        with_payload: Some(true.into()),
-                        params: None,
-                        score_threshold: None,
-                        offset: None,
-                        vector_name: None,
-                        with_vectors: None,
-                        read_consistency: None,
-                    })
-                    .await?;
-
-                utils::done(&format!(
-                    "DSSTAT Finished searching Qdrant documents: collection_name={}, duration={}ms, results_count={}",
-                    self.qdrant_collection(),
-                    utils::now() - time_search_start,
-                    results.result.len(),
-                ));
-
-                let time_chunk_start = utils::now();
-                let chunks = parse_points_into_chunks(
-                    &(results
-                        .result
-                        .into_iter()
-                        .map(QdrantPoint::Scored)
-                        .collect()),
-                )?;
-
-                utils::done(&format!(
-                    "DSSTAT Finished chunking documents: collection_name={}, duration={}ms, chunk_length={}",
-                    self.qdrant_collection(),
-                    utils::now() - time_chunk_start,
-                    chunks.len(),
-                ));
+                match q.len() {
```

Comment:
I'm surprised you repeat the code here. Let's just do right before something like

```
let query = match(query) {
  Some(q) =>  match q.len() {
    0 => None,
    _ => Some(q)
  },
  None => None
}
```

Reply:
This is indeed way better... I was tired 

Fixed !